source/agora/test/Base.d

```diff
@@ -796,8 +796,8 @@ public class TestAPIManager
                     scope node = this.nodes[idx];
                     retryFor(node.client.getPreimages(Set!Hash.from(enroll.utxo_key.only))
                         .any!(preimage => preimage.height >= height),
-                            timeout, format!"Client #%s (%s) has no preimage for client #%s (%s) at height %s"
-                                (idx, node.address, idx_enroll, this.nodes[idx_enroll].address, height));
```

Comment:
The `this.nodes[idx_enroll].address` is an important information about a validator for which the client is missing pre-images. Can we just leave it as it was if there's so big problem?

Reply:
Problem is that the index of the enrollment can be different to the index of the node, in this case the wrong address is logged. I saw it happen for `ValidatorRecurringEnrollment` test.

Reply:
> > Previously the index of the enrollment was logged whicj may not be the
> > actual index of the node. To be always accurate the utxo is logged.
> 
> In which situation ?

I updated the commit description to give an example.